src/main/app/claims/searchClaim.ts

```diff
@@ -0,0 +1,29 @@
+import Claim from 'claims/models/claim'
+import ClaimStoreClient from 'claims/claimStoreClient'
+
+export class SearchClaim {
+
+  static search (query: string, userAuthToken: string): Promise<Claim> {
+    if (!query) {
+      return Promise.reject(new Error('Search text cannot be blank'))
+    }
+    if (!userAuthToken) {
+      return Promise.reject(new Error('User auth token cannot be blank'))
+    }
+    const text: string = query.trim()
+    if (SearchClaim.isClaimReferenceNumber(text)) {
```

Comment:
this looks a little too magical to me, I think it would be better to separate the queries, what do you think?

Reply:
The prototype has single text field to search, so I am afraid we have to implement this magic.